lib/query.js

```diff
@@ -3353,6 +3355,11 @@ Query.prototype.findOneAndUpdate = function(filter, doc, options) {
     throw new MongooseError('Query.prototype.findOneAndUpdate() no longer accepts a callback');
   }
 
+  // Check for empty filter with strictFilter option
```

Comment:
I don't think this is the right place for this check because of query chaining. For example, `await TestModel.findOneAndUpdate({}, { isPublic: true }).setQuery({ isPublic: false })` will execute with a non-empty filter even though the original `findOneAndUpdate()` call had an empty filter.